station/recipes/factory.py

```diff
@@ -1,81 +1,72 @@
+* Frequency - it is frequency for SDR.
+* Duration - after this time script should stop SDR
 
-As output the script should print on standard output in format:
-    !! CATEGORY: PATH
-where CATEGORY is category of file. Now I use "Signal" for initial WAV file and
-"Product" for decoded imagery. I think that we will need a category for waterfall
-and text, binary data in future.
+As output the script returns a dictionary where key is a category
+and value is a path or paths to file(s).
+Now we use "signal" for initial WAV file and
+"product" or "products" for decoded imagery.
+I think that we will need a category for waterfall
+and telemetry (raw and/or decoded) in future.
 
-User should have a possibility to execute recipe without using Python code.
+User should have a possibility to execute recipe with separation of station core.
+(Without need to initialize the station).
 
-Recipe is responsible for cleanup all created files excluded the files printed
-as output. These files are clean up by Python code.
+Recipe isn't responsible for clean up. The working directory should be deleted
+after file proceeding. If some files should be keep then they should be moved
+in safe place.
 
 User has a possibility to select recipe by "recipe" parameter in satellite
 config. Now it is optional and script try to deduce recipe based on satellite. 
 (For example if recipe isn't provided and name starts with NOAA then we use
 recipe for NOAA).
 
-All recipes must be located in directory with this file ("recipes"), have ".sh"
-extensions and set execute rights for owner.
-
-Your shell script should starts with the shebang (e.g. #!/bin/sh).
+All recipes must be located in directory with this file ("recipes") and have
+"execute" method which accept all parameters.
 '''
 
-RECIPE_DIR = os.path.split(os.path.abspath(__file__))[0]
 BASE_DIR = "/tmp/observations_tmp"
 os.makedirs(BASE_DIR, exist_ok=True)
 
-def get_recipe(sat: SatelliteConfiguration) -> str:
+def get_recipe(sat: SatelliteConfiguration):
     '''
     Returns path to recipe file assigned with passed satellite.
     If recipe doesn't exist throw LookupError.
     
     Function check "recipe" field in passed configuration for recipe.
     If it is empty then check built-in list with compatible recipes.
     '''
+    recipe = None
     if "recipe" in sat:
         recipe = sat["recipe"]
-    elif sat["name"].startswith("NOAA"):
+    if recipe is None and sat["name"].startswith("NOAA"):
         recipe = "noaa-apt"
-    else:
+    if recipe is None:
         raise LookupError("Unknown recipe")
 
-    recipe_filename = os.path.join(RECIPE_DIR, recipe + ".sh")
-    if os.path.exists(recipe_filename):
-        return recipe_filename
-    else:
-        raise LookupError("Recipe doesn't exist")
+    return recipes[recipe]
+
 
 def execute_recipe(sat: SatelliteConfiguration, los: datetime.datetime) -> Iterable[Tuple[Literal["signal", "product"], str]]:
```

Comment:
I think we should not limit the type of returned categories to just "signal" and "product". There likely be more types in a near future (waterfall, telemetry, and logs are good examples).

Reply:
We don't have any "hard" limitation of categories. It is only typing literal - if we use in code any unknown category then we should get a warning (in practice probably no...).

Reply:
I added the comment.

Reply:
LGTM